packages/sdk/test/integration/PublisherKeyExchange.test.ts

```diff
@@ -67,63 +89,100 @@ describe('PublisherKeyExchange', () => {
-        await publisherClient.updateEncryptionKey({
-            key,
-            distributionMethod: 'rekey',
-            streamId: StreamPartIDUtils.getStreamID(streamPartId)
-        })
-
-        await triggerGroupKeyRequest(erc1271ContractAddress)
+        await triggerGroupKeyRequest(publisherClient, stream, {}, erc1271ContractAddress)
 
         const response = await environment.getNetwork().waitForSentMessage({
             messageType: StreamMessageType.GROUP_KEY_RESPONSE
         })
-        await assertValidResponse(response, key, toUserId(erc1271ContractAddress), SignatureType.ERC_1271)
+        await assertValidResponse(response, stream.id, key, toUserId(erc1271ContractAddress), SignatureType.ERC_1271)
+    })
+
+    describe('quantum resistant key exchange', () => {
+        it('works when both publisher and subscriber require quantum security', async () => {
+            const { publisherClient, stream } = await createClientAndStream({
+                encryption: {
+                    requireQuantumResistantKeyExchange: true,
+                }
+            })
+    
+            await triggerGroupKeyRequest(publisherClient, stream, {
+                encryption: {
+                    requireQuantumResistantKeyExchange: true,
+                }
+            })
+    
+            const response = await environment.getNetwork().waitForSentMessage({
+                messageType: StreamMessageType.GROUP_KEY_RESPONSE
+            })
+            await assertValidResponse(response, stream.id, key, toUserId(publisherWallet.address), SignatureType.SECP256K1)
+        })
+
+        it('works when subscriber requests quantum security', async () => {
+            const { publisherClient, stream } = await createClientAndStream()
+    
+            await triggerGroupKeyRequest(publisherClient, stream, {
+                encryption: {
+                    requireQuantumResistantKeyExchange: true,
+                }
+            })
+    
+            const response = await environment.getNetwork().waitForSentMessage({
+                messageType: StreamMessageType.GROUP_KEY_RESPONSE
+            })
+            await assertValidResponse(response, stream.id, key, toUserId(publisherWallet.address), SignatureType.SECP256K1)
+        })
+
+        it('fails when publisher requires it but subscriber does not', async () => {
+            const { publisherClient, stream } = await createClientAndStream({
+                encryption: {
+                    requireQuantumResistantKeyExchange: true,
+                }
+            })
+    
+            // The invalid request just prints a debug level error on the publisher.
+            // Since the error is swallowed, it's difficult to assert the error here.
+            await triggerGroupKeyRequest(publisherClient, stream)
+    
+            // Due to the above, rely on time - not foolproof
+            await new Promise((resolve) => setTimeout(resolve, 3000))
```

Comment:
Could use wait from `@streamr/utils`

```suggestion
            await wait(3000)
```